### PR TITLE
peci: driver: Ping command failing due to incorrect tx length handling and wait timeout

### PR DESCRIPTION
**Pull request by DiwakarCC**
**Bug description**

Below 2 issues has to be fixed:

Issue 1: Ping command have write and read length of '0'. In current implementation, when ping command is sent to driver, payload of size (Tx_Buf_Len-1) is added to the FIFO through below loop.

for (int i=0; i<(Tx_Buf_Len-1); i))
{
// Add to FIFO
}

This works for all PECI commands except for ping command. Incase of ping command, Tx_Buf_Len=0 results in comparison (0 < -1(0xFFFFFFFF)) which is comparison b/w signed and unsigned numbers. When signed (index i) and unsigned (Tx_Buf_Len) compared, compiler implicitly converts signed to unsigned results in entering infinite loop (since 0<0xFFFFFFFF is true).

Issue 2: wait_timeout never decrements hence the loop is waiting infinitely for (!(base->STATUS1 & MCHP_PECI_STS1_EOF)) to become true. 

**Steps to reproduce:**
Send peci_ping() command to peci driver.

**Expected behavior:**
Ping command tx should success.

**Impact:**
Peci app will fail for ping command.

### CODE REVIEW

File: drivers/peci/peci_mchp_xec.c [hunk 1]
```diff
@@ -175,11 +170,13 @@ static int peci_xec_write(const struct device *dev, struct peci_msg *msg)
 		return -ETIMEDOUT;
 	}
 #else
-	wait_timeout = tx_buf->len;
+	/* In worst case, overall timeout will be 1msec (100 * 10usec) */
+	u8_t wait_timeout_cnt = 100;
```
Comment (albertofloyd) [inline drivers/peci/peci_mchp_xec.c:174]: Need to update types uint8

Comment (franciscomunoz) [reply drivers/peci/peci_mchp_xec.c:174]: just fix the types everything else looks ok

**Comment by albertofloyd** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/pull/29708

**Comment by DiwakarCC** [comment]
Close

**Pull request closed without merging**